### PR TITLE
add wrapper of forward

### PR DESCRIPTION
**Pull request by msdsm**
# Add `forward_by_train()` and `forward_by_eval()` convenience methods to nn.Module

## Summary

Adds two new methods to `nn.Module` that automatically set the training mode before executing forward pass:
- `forward_by_train(*args, **kwargs)` - calls `train()` then forwards
- `forward_by_eval(*args, **kwargs)` - calls `eval()` then forwards

## Motivation

Users sometimes forget to call `model.train()` or `model.eval()` before inference, leading to incorrect behavior with modules like `Dropout` and `BatchNorm`. These methods make the intent explicit and prevent such errors.

## Implementation

```python
def forward_by_train(self, *args: Any, **kwargs: Any) -> Any:
    self.train()
    return self(*args, **kwargs)

def forward_by_eval(self, *args: Any, **kwargs: Any) -> Any:
    self.eval()
    return self(*args, **kwargs)
```

## Example Usage

```python
# Before
model.eval()
with torch.no_grad():
    output = model(input)

# After
with torch.no_grad():
    output = model.forward_by_eval(input)
```

### CODE REVIEW

**Comment by albanD** [comment]
Thank you very much for sending a PR.
To avoid you spending time preparing a PR for a feature we're not sure we want to add, our usual process is to discuss on the issue before doing the PR.
See https://github.com/pytorch/pytorch/wiki/The-Ultimate-Guide-to-PyTorch-Contributions for details!

In this case, I would suggest oppening an issue with the details and discuss the API design there.

**Comment by msdsm** [comment]
I’ve opened an issue for discussion here: #170851
